src/api/tasks.py

```diff
@@ -328,14 +331,59 @@ def automatically_activate_nudge(community_nudge_setting_id):
             community_nudge_setting.is_active = True
             community_nudge_setting.save()
             
-            logging.info(f"Successfully activated nudge for community: {community_nudge_setting.community.name}")
+            log.info(f"Successfully activated nudge for community: {community_nudge_setting.community.name}")
             
         except Exception as e:
-            logging.error(f"Error automatically activating nudge: {str(e)}")
+            log.exception(e)
             
         finally:
-            logging.info(f"===== Deleting Cache Key: {cache_key} =====")
+            log.info(f"===== Deleting Cache Key: {cache_key} =====")
             cache.delete(cache_key)
+    else:
+        log.info("Task already picked up by another worker")
+        
+        
+@app.task
+def update_recurring_event(event_id):
+    
+    cache_key = f"update_recurring_event_{event_id}"
+    
+    if cache.add(cache_key, True, timeout=3600):
```

Comment:
i dont think this is the right way to use a cache.
I think we should create a task with a name = the event's id or something and leverage the transaction logic we have for processing that one off task that runs at some future time = the event's end datetime